### PR TITLE
feat: add metrics support to the sink path

### PR DESCRIPTION
**Pull request by ali-ince**

### CODE REVIEW

File: common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      DataWriterMetric(PROPERTIES_SET, propertiesSet.longValue()),
+      DataWriterMetric(LABELS_ADDED, labelsAdded.longValue()),
+      DataWriterMetric(LABELS_REMOVED, labelsRemoved.longValue())
+    ).toArray
+  }
+
+}
+
+object DataWriterMetrics {
+  final val RECORDS_WRITTEN = "recordsWritten"
```
Comment (venikkin) [inline common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala]: I don't know that are the naming conventions spark have for metrics, but looking at existing (for instance https://spark.apache.org/docs/latest/monitoring.html#executor-metrics), spark sometimes combines them by groups like `memoryMetrics.*` or `peakMemoryMetrics`. I wonder if it makes sense to prefix our custom metrics with something like `neo4jMetrics.*`?

**Review by venikkin — APPROVED** [review]
Looks good. Have small suggestion / question reg the metrics naming.

File: common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala (hunk 1 above)
Comment (fbiville) [reply common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala]: I like this idea. The only potential gotcha is that the name of custom metrics that seem to surface are based on the descriptions we provide, not the name.
WDYT @ali-ince?

Comment (ali-ince) [reply common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala]: Yeah, I agree that it's a nice idea. Let's use namespaced naming scheme as @venikkin suggests, and we can update the descriptions in the future if we need to.

Comment (fbiville) [reply common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala]: sorry, a bit slow on that one, but here is the change: c9d214d5f8f325661d34b4944f5bddb5b5a9190a.
what do you both think?

**Review by venikkin — APPROVED** [review]

**Pull request merged**